pkg/clusteragent/admission/mutate/common.go

```diff
@@ -153,3 +159,67 @@ func containsVolumeMount(volumeMounts []corev1.VolumeMount, element corev1.Volum
 	}
 	return false
 }
+
+// shouldInject returns true if Admission Controller should inject standard tags, APM configs and APM libraries
+func shouldInject(pod *corev1.Pod) bool {
```

Comment:
It looks like with this function that APM libraries will be injected if `mutate_unlabelled` is `true` which I think we wanted to avoid

Reply:
We discussed on Zoom, but will reply here for reference and visibility.
`shouldInject(...)` is used to decide if tags and APM configs need to be injected, thus we are checking all three - label, mutateUnlabelled and SSI configuration.

The next PR adds another method [shouldMutatePod(...)](https://github.com/DataDog/datadog-agent/pull/19595/files#diff-5d1f23f8fcec88188fb13bade49122f4cff59445aa97c64da3b84639d8355040R165) that doesn't take into account SSI configuration. To make a decision if the libraries need to be injected, we use method `isApmInstrumentationEnabled(...)` that doesn't consider mutateUnlabelled.

Reply:
the APM library case is handled separately in #19595 so this isn't an issue